src/app/auth/session-selection/session-selection.component.ts

```diff
@@ -117,6 +115,11 @@ export class SessionSelectionComponent implements OnInit {
   }
 
   private autofillRepo() {
-    this.repoForm.get('repo').setValue(this.urlEncodedRepo);
+    const repoLocation = this.cacheRepoFromUrlService.repoLocation || this.urlEncodedRepo;
+    this.repoForm.get('repo').setValue(repoLocation);
+
+    if (this.cacheRepoFromUrlService.hasRepoLocation()) {
+      this.setupSession();
+    }
```

Comment:
From this, we only set up a session if `cacheRepoFromUrlService.repoLocation != null`

But setupSession does not use `cacheRepoFromUrlService`, and it will never set up the session if we are using the url encoded repo.